sdk/apimanagement/Azure.ResourceManager.ApiManagement/src/Customized/Models/ApiCreateOrUpdateContent.Serialization.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ClientModel.Primitives;
+using System.Runtime.CompilerServices;
+using System.Text.Json;
+using Azure.Core;
+
+namespace Azure.ResourceManager.ApiManagement;
+
+[CodeGenSerialization(nameof(TermsOfServiceUri), DeserializationValueHook = nameof(DeserializetermsOfServiceUrlValue))]
+public partial class ApiData : IUtf8JsonSerializable, IJsonModel<ApiData>
+{
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    internal static void DeserializetermsOfServiceUrlValue(JsonProperty property0, ref Uri termsOfServiceUrl)
+    {
+        if (property0.Value.ValueKind == JsonValueKind.Null)
+            return;
+        var str = property0.Value.GetString();
+        if (!string.IsNullOrEmpty(str))
+        {
+            if (Uri.TryCreate(str, UriKind.Absolute, out _))
+            {
+                termsOfServiceUrl = new Uri(str);
+            }
```

Comment:
I do not think we should fix this issue in this way. This breaks the roundtrip of serialization and deserialization.
Imagine we have a json `{"termOfServiceUri": "some invalid uri"}`, we first deserialize this into the model, we get a model with null value in `TermOfServiceUri` property. And if we serialize it again, the value of that property just disappears.
We should not do this.

Instead, I believe we should introduce a new `string` property `TermOfService` or `TermOfServiceLink` (maybe, we could discuss about the name), and introduce the `Uri` property back using the TryCreate to create a value from the string property as its backing field.

Reply:
same comment applies to the other property.